dali/operators/image/color/color_space_conversion.h

```diff
@@ -26,23 +26,47 @@ template <typename Backend>
 class ColorSpaceConversion : public Operator<Backend> {
  public:
   inline explicit ColorSpaceConversion(const OpSpec &spec)
-    : Operator<Backend>(spec)
-    , input_type_(spec.GetArgument<DALIImageType>("image_type"))
-    , output_type_(spec.GetArgument<DALIImageType>("output_type")) {
+      : Operator<Backend>(spec),
+        input_type_(spec.GetArgument<DALIImageType>("image_type")),
+        output_type_(spec.GetArgument<DALIImageType>("output_type")),
+        in_nchannels_(NumberOfChannels(input_type_)),
+        out_nchannels_(NumberOfChannels(output_type_)) {
   }
 
  protected:
-  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
-    return false;
+  bool CanInferOutputs() const override { return true; }
+  bool SetupImpl(std::vector<OutputDesc> &output_desc,
+                 const workspace_t<Backend> &ws) override {
+    output_desc.resize(1);
+    const auto &input = ws.template InputRef<Backend>(0);
+    auto in_sh = input.shape();
+    auto ndim = in_sh.sample_dim();
+    int nsamples = in_sh.num_samples();
+    auto in_layout = input.GetLayout();
+    int channel_dim = ndim - 1;
+    DALI_ENFORCE(IsType<uint8_t>(input.type()), "Color space conversion accept only uint8 tensors");
+    DALI_ENFORCE(
+        in_layout.empty() || in_layout.find('C') == channel_dim,
+        make_string("Channel dimension should be the last in the layout. Got ", in_layout));
```

Comment:
The layouts are listed in the schema - all of them have trailing channel. Remove the check here or remove the list of layouts from the schema - whichever suits you better.

Reply:
I'll remove this